src/main/java/net/countercraft/movecraft/combat/features/tracking/BlockBehaviorOverride.java

```diff
@@ -0,0 +1,420 @@
+            } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException
+                     | NoSuchMethodException | SecurityException | ClassNotFoundException e) {
+                e.printStackTrace();
+                return false;
+            }
+        }
+
+        public Optional<Integer> getBurnOdds(Material m) {
+            try {
+                Object fireBlock = this.getBlockClass(Material.FIRE);
+                Object block = this.getBlockClass(m);
+                Optional<Object2IntMap> optMap = getFieldValueSafe(fireBlock, this.fieldNameBurnOdds);
+                if (optMap.isPresent()) {
+                    if (optMap.get().containsKey(block)) {
+                        return Optional.ofNullable(optMap.get().getInt(block));
+                    }
+                    return Optional.empty();
+                }
+            } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException
+                     | NoSuchMethodException | SecurityException | ClassNotFoundException e) {
+                e.printStackTrace();
+            }
+            return Optional.empty();
+        }
+
+        public boolean setIgniteOdds(Material m, int value) {
+            try {
+                Object fireBlock = this.getBlockClass(Material.FIRE);
+                return setIgniteOdds(m, value, fireBlock);
+            } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException
+                     | NoSuchMethodException | SecurityException | ClassNotFoundException e) {
+                e.printStackTrace();
+                return false;
+            }
+        }
+
+        public Optional<Integer> getIgniteOdds(Material m) {
+            try {
+                Object fireBlock = this.getBlockClass(Material.FIRE);
+                Object block = this.getBlockClass(m);
+                Optional<Object2IntMap> optMap = getFieldValueSafe(fireBlock, this.fieldNameIgniteOdds);
+                if (optMap.isPresent()) {
+                    if (optMap.get().containsKey(block)) {
+                        return Optional.ofNullable(optMap.get().getInt(block));
+                    }
+                    return Optional.empty();
+                }
+            } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException
+                     | NoSuchMethodException | SecurityException | ClassNotFoundException e) {
+                e.printStackTrace();
+            }
+            return Optional.empty();
+        }
+
+        protected boolean setBurnOdds(Material m, int value, Object fireBlock) {
+            try {
+                final Object block = getBlockClass(m);
+                // First Object2Identiy map in field list is the one for ignite odds, second one is for burn odds
+                Consumer<Object2IntMap> func = (map) -> {
+                    map.put(block, value);
+                };
```

Comment:
This functional lambda can be replaced with an expression lambda: `(map) -> map.put(block, value)`